test_vm/tests/verified_claim_test.rs

```diff
@@ -298,3 +300,94 @@ fn verified_claim_scenario() {
+
+    // Register verifier and verified clients
+    let datacap = StoragePower::from(32_u128 << 40);
+    verifreg_add_verifier(&v, verifier, &datacap * 2);
+    verifreg_add_client(&v, verifier, verified_client, datacap.clone());
+
+    // Add market collateral for client and miner
+    market_add_balance(&v, verified_client, verified_client, TokenAmount::from_whole(3));
+    market_add_balance(&v, worker, miner_id, TokenAmount::from_whole(64));
+
+    // Publish 2 verified deals
+    let deal1_start =
+        v.get_epoch() + max_prove_commit_duration(&Policy::default(), seal_proof).unwrap();
+    let deal_term_min = 180 * EPOCHS_IN_DAY;
+
+    let deal_size = 32u64 << 30;
+    let deal1 = market_publish_deal(
+        &v,
+        worker,
+        verified_client,
+        miner_id,
+        "deal1".to_string(),
+        PaddedPieceSize(deal_size),
+        true,
+        deal1_start,
+        deal_term_min,
+    )
+    .ids[0];
+
+    // Client datacap balance reduced
+    let datacap_state: DatacapState = v.get_state(DATACAP_TOKEN_ACTOR_ADDR).unwrap();
+    assert_eq!(
+        TokenAmount::from_whole(datacap.clone()) - TokenAmount::from_whole(deal_size),
+        datacap_state.token.get_balance(&v.store, verified_client.id().unwrap()).unwrap()
+    );
+
+    // Advance to after the first deal's start
+    let v = v.with_epoch(deal1_start + EPOCHS_IN_DAY);
+    cron_tick(&v);
+
+    // Deal has expired and cleaned up.
+    let market_state: MarketState = v.get_state(STORAGE_MARKET_ACTOR_ADDR).unwrap();
+    let proposals: DealArray<MemoryBlockstore> =
+        DealArray::load(&market_state.proposals, v.store).unwrap();
+    assert!(proposals.get(deal1).unwrap().is_none());
+    let pending_deal_allocs: Map<MemoryBlockstore, AllocationID> = make_map_with_root_and_bitwidth(
+        &market_state.pending_deal_allocation_ids,
+        v.store,
+        HAMT_BIT_WIDTH,
+    )
+    .unwrap();
+    assert!(pending_deal_allocs.get(&deal_id_key(deal1)).unwrap().is_none());
+
+    // Allocation still exists until explicit cleanup
+    let alloc_id = 1;
+    let verifreg_state: VerifregState = v.get_state(VERIFIED_REGISTRY_ACTOR_ADDR).unwrap();
+    let mut allocs = verifreg_state.load_allocs(v.store).unwrap();
+    assert!(allocs.get(verified_client.id().unwrap(), alloc_id).unwrap().is_some());
+
+    verifreg_remove_expired_allocations(&v, worker, verified_client, vec![], deal_size);
+
+    // Allocation is gone
+    let verifreg_state: VerifregState = v.get_state(VERIFIED_REGISTRY_ACTOR_ADDR).unwrap();
+    let mut allocs = verifreg_state.load_allocs(v.store).unwrap();
+    assert!(allocs.get(verified_client.id().unwrap(), alloc_id).unwrap().is_none());
+
+    // Client has original datacap balance
+    let datacap_state: DatacapState = v.get_state(DATACAP_TOKEN_ACTOR_ADDR).unwrap();
+    assert_eq!(
+        TokenAmount::from_whole(datacap),
+        datacap_state.token.get_balance(&v.store, verified_client.id().unwrap()).unwrap()
+    );
```

Comment:
extra: your last pr reminded me, if it doesn't take too long a call to `v.check_state_invariants` would be nice here.